### DISCUSSION TITLE
What casing do you use for your collection names?

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#2074] [Polls]
I wonder how to name collections that consist of multiple words, e.g. "blog posts". Here's a relevant [question on Stack Overflow about naming conventions for MongoDB collections](https://stackoverflow.com/q/5916080/3130281).

I can see that Payload appears to use underscores internally, judging by the system generated `_*_versions` collection, where `*` is the slug of your collection.

**Comment by yandodov** [comment]
I think I like kebab-case the most, because URLs make more sense:

- http://localhost:3000/admin/collections/blog_posts is kind of odd
- http://localhost:3000/admin/collections/blog-posts looks great

It also establishes a visual hierarchy with Payload collections. For example, in `_blog-posts_versions` the dash indicates that "posts" is just part of the collection name, while the underscore indicates that "versions" is kind of a property of the collection, like how you would use a dot in JavaScript to do the same thing, i.e. `blogPosts.versions`. On the other hand, `_blog_posts_versions` establishes no such hierarchy.

**Reply by thgh** [reply]
I would say camel > snake > kebab.
Camelcase because it's idiomatic in JS. And snakecase over kebab because having `collections['blog-posts']` is less awesome than `collections.blog_posts`. When double clicking to select a keyword, the OS handles underscore as part of the word while the dash will split it up. Also, underscore looks more like a space than a dash. To finish it off, wikipedia uses _ in URLs!

**Reply by yandodov** [reply]
Hm, valid points! However, now that I've looked at it, [the official Payload site uses kebab](https://github.com/payloadcms/website-cms/blob/2e4f20b29317c849e41488e94fcc6608488f1630/src/collections/CaseStudies.ts#L26). I wonder if they've deliberately chosen that.

**Comment by yandodov** [comment]
I just found out about an issue in MongoDB shell:

> If your collection name includes special characters, such as the underscore character, or begins with numbers, then to access the collection use the [db.getCollection()](https://docs.mongodb.com/manual/reference/method/db.getCollection/#db.getCollection) method in the mongo shell or a similar method for your driver.

- [Stack Overflow question](https://stackoverflow.com/questions/24309685/mongo-client-cant-access-collections-prefixed-with-an-underscore)
- [MongoDB Jira issue SERVER-445](https://jira.mongodb.org/browse/SERVER-445)

This can't be quite annoying.

@jmikrut @DanRibbens perhaps that's a reason to avoid underscores both as a user *and* internally in Payload? For example, I have issues querying my `_blog-posts_versions` collection, because it starts with an underscore.

The prefix could simply be `payload_`, i.e. `payload_blog-posts_versions`, `payload_preferences`, etc.
